lib/std/priority_queue.zig

```diff
@@ -190,6 +190,31 @@ pub fn PriorityQueue(comptime T: type) type {
             self.len = new_len;
         }
 
+        fn binarySearch(items: []const T, target: T) usize {
```

Comment:
`std.sort.binarySearch`

Reply:
oh ok, will change that

Reply:
Beside the wrong use of the binary search function and the fact it's not using the PriorityQueue `compareFn` I'd suggest to have a look at how the elements are ordered inside the `items` array.
A cursory look shows that they're not ordered (even though I expected them to be) so using the binary search here is not possible.
Moreover there's a more efficient way of performing the update (or increase/decrease), see [here](https://en.wikipedia.org/wiki/Binary_heap#Decrease_or_increase_key).

Reply:
"the fact it's not using the PriorityQueue compareFn" => wouldn't need that to find the element.

Okay, so I would need to first "Find the index of the element we want to modify"; in which case I would need to either sort the items first, and use binary search or just use linear search.

Moreover there's a more efficient way of performing the update (or increase/decrease) => heapreplace, will try that out; let me get it working with linear search + remove and add first

Reply:
> "the fact it's not using the PriorityQueue compareFn" => wouldn't need that to find the element.

If the elements were sorted the same comparison had to be used in the binary search.

> in which case I would need to either sort the items first, and use binary search or just use linear search.

Don't bother with sorting, linear search will do.

Reply:
> If the elements were sorted the same comparison had to be used in the binary search.

yes, okay, however in this case just linear search would be more efficient